srtcore/packet.cpp

```diff
@@ -194,9 +194,23 @@ m_pcData((char*&)(m_PacketVector[PV_DATA].dataRef()))
     m_PacketVector[PV_DATA].set(NULL, 0);
 }
 
+char* CPacket::getData()
+{
+    return (char*)m_PacketVector[PV_DATA].dataRef();
+}
+
 void CPacket::allocate(size_t alloc_buffer_size)
 {
+    if (m_data_owned)
+    {
+        if (getLength() == alloc_buffer_size)
+            return; // already allocated
+
+        // Would be nice to reallocate; for now just allocate again.
+        delete [] m_pcData;
+    }
     m_PacketVector[PV_DATA].set(new char[alloc_buffer_size], alloc_buffer_size);
+    setLength(alloc_buffer_size);
     m_data_owned = true;
```

Comment:
the `set()` function already sets the length. So `setLength(alloc_buffer_size)` is redundant.

Reply:
Right. Likely this is an old function created before you made this `IOVector` stuff.